web/src/settings_preferences.ts

```diff
@@ -254,6 +254,15 @@ export function set_up(settings_panel: SettingsPanel): void {
                     : NON_COMPACT_MODE_LINE_HEIGHT_PERCENT;
             }
 
+            if (
+                ((setting === "web_font_size_px" && setting_value !== LEGACY_FONT_SIZE_PX) ||
+                    (setting === "web_line_height_percent" &&
+                        setting_value !== LEGACY_LINE_HEIGHT_PERCENT)) &&
+                user_settings.dense_mode
+            ) {
+                data.dense_mode = false;
```

Comment:
This is required as we show the inputs always in dev environment and we allow changing only one setting in a request.

I think we can hide the inputs if "Compact mode" is checked in dev environment as well and then show when it is unchecked.

Reply:
I'm not sure how important the development-environment specific logic is; my guess is we'll be opening it up in `main` within a month or so anyway.